rdkit/Chem/Draw/IPythonConsoleIntegration.py

```diff
@@ -0,0 +1,558 @@
+    b.icon='check'
+    b.description="Done!"
+    uid=globals()['wgMIds'][b.model_id]
+    globals()['viewInstantiated'][uid] = True
+    update3D(b.model_id)
+    
+def handle_zoomTo_button(b):
+    """This function handles zoomTo button"""
+    uid=globals()['wgMIds'][b.model_id]
+    wgData = globals()['wgData'][uid]
+    wgData['view'].zoomTo()
+    display(wgData['view'].update())
+    
+    
+def ChangeActiveLigand(uid, molId, confId, removeAllModels = False):
+    """This function handles ligand select through python code
+    uid : string 
+    molId : string 
+    confId : intiger
+    removeAllModels : logical (default False,  if True - all models removed)
+    """
+    
+    if isinstance(uid, str if PY3 else basestring) is False:
+        raise TypeError("uid must be a string")
+        
+    try:
+        wgData = globals()['wgData'][uid]
+    except:
+        raise KeyError("invalid uid")
+        
+    if isinstance(molId, str if PY3 else basestring) is False:
+        raise TypeError("molId must be a string")
+        
+    if molId not in globals()['molData'][uid].moldict:
+        raise KeyError("invalid molId")
+        
+    # TODO: handle invalid confId
+    
+    wgData['molId'].value = molId
+    wgData['confId'].value = confId
+    
+    
+    if removeAllModels:
+        wgData['selectMultiMols'].value=False
+        wgData['selectAllMols'].value=False
+        wgData['selectMultiConfs'].value=False
+        wgData['selectAllConfs'].value=False
+    
+    update3D(wgData['molId'].model_id)
+    
+    
+def ShowConformers3D(uid = None,
+                     moldict = None, protein = None,
+                     useDrawAs = False, 
+                     drawAs='stick', pStyle='cartoon',
+                     propPanel = False, calculateProperties = False,
+                     colorPanel = False, 
+                     labelPanel = False):
+    
+    """This function initiates required widgets and 3Dmol.js viewer
+    uid : an unique id for 3DMol.js viewer, each of the ipywidgets, and all the global objects
+    moldict : it is a molecule dictionary where is keys are string
+    protein : it is rdkit standard mol object
+    drawAs : representation type for moldict or small molecule
+    pStyle : protein representation style
+    ** the rest of the arguments are panels (user may or may not need those)
+    """
+    
+    if uid is None:
+        uid = str(time.time()).replace('.','')
+        
+    if isinstance(uid, str if PY3 else basestring) is False:
+        raise TypeError("uid must be str")
+        
+    if isinstance(moldict, dict) is False:
+        raise TypeError("moldict must be dict")
+        
+    if all(isinstance(key, str if PY3 else basestring) for key in moldict.keys()) is False:
+        raise TypeError("keys of moldict must be str")
+        
+    wgMIds = globals().setdefault('wgMIds', {})
+    
+    wgData = globals().setdefault('wgData', {})
+    wgData[uid] = {}
```

Comment:
wgMIds dependency removed. So, I am taking your comments as for my lesson. I hope you will like the most recent codes. Thank you.